ports/stm/common-hal/microcontroller/Pin.c

```diff
@@ -89,6 +89,8 @@ void reset_pin_number(uint8_t pin_port, uint8_t pin_number) {
 
 void never_reset_pin_number(uint8_t pin_port, uint8_t pin_number) {
     never_reset_pins[pin_port] |= 1<<pin_number;
+    // Make sure never reset pins are also always claimed
+    claimed_pins[pin_port] |= 1<<pin_number;
```

Comment:
The other ports don't do this.  I think it would be better to either
 * Add it to all ports, and make it a part of the contract of `never_reset_pin_number` that it claims the pin
   * if so, maybe there's an overall code size savings if `claim_pin` calls can be folded into `never_reset` calls.
 * or `assert(claimed_pins[pin_port] & (1<<pin_number)` so that mistakes are caught as soon as possible.
   * again, probably better to add the assertion on all ports
   * requires developers to test with DEBUG=1 builds

Reply:
Yeah this one was kind of frustrating, because it's really an API issue: across all ports, `claim_pin` is always called with a pin object as the parameter, but `never_reset_pin_number` is called in the context of a reset function. Reset functions don't actually have pin objects, they just iterate through port and mux numbers, so I can't call claim_pin without doing a wasteful lookup for the correct pin object. 

I should probably just make another alias for claim_pin that uses the port and number instead... I'll see how much impact that has.